nbgrader/apps/api.py

```diff
@@ -979,3 +979,104 @@ def autograde(self, assignment_id, student_id, force=True, create=True):
             app.force = force
             app.create_student = create
             return capture_log(app)
+
+    def generate_feedback(self, assignment_id, student_id=None):
+        """Run ``nbgrader generate_feedback`` for a particular assignment and student.
+
+        Arguments
+        ---------
+        assignment_id: string
+            The name of the assignment
+        student_id: string
+            The name of the student (optional). If not provided, then generate
+            feedback from autograded submissions.
+
+        Returns
+        -------
+        result: dict
+            A dictionary with the following keys (error and log may or may not be present):
+
+            - success (bool): whether or not the operation completed successfully
+            - error (string): formatted traceback
+            - log (string): captured log output
+
+        """
+        if student_id is not None:
+            with temp_attrs(self.coursedir,
+                            assignment_id=assignment_id,
+                            student_id=student_id):
+                app = Feedback(coursedir=self.coursedir, parent=self)
+                c = Config()
+                c.HTMLExporter.template_file = 'feedback.tpl'
```

Comment:
@dsblank should be able to comment on this one.

Reply:
The issue here was that the HTMLExporter was being used in other place(s) in conversion. The code in the Feedback converter would only set the template to "feedback.tpl" if the template had not  already been set in the config. But it had been set, so we have to manually set it here to override that. (Hmmm, I wonder if we need to set it back...?)

Reply:
Thanks for clarifying---can you add a comment in the code with this explanation?

> I wonder if we need to set it back...?

I think probably not, since it creates a new config object on the fly here.

Reply:
Added comments and refactor in https://github.com/danielmaitre/nbgrader/pull/8